### PR TITLE
chore(reveal-react-components): add formatting and linting

### PR DESCRIPTION
**Pull request by christjt**
#### Type of change

Chore 

## Description :pencil:

Add prettier and eslint for formatting and linting

### CODE REVIEW

File: react-components/tsconfig.json [hunk 1]
```diff
@@ -5,19 +5,14 @@
...
-  "include": [
-    "src"
-  ],
-  "exclude": [
-    "**/*.stories.ts"
-  ]
-}
\ No newline at end of file
+  "include": ["src/*", "stories/*"]
+}
```
Comment (haakonflatval-cognite) [inline react-components/tsconfig.json:18]: Is this right? It seems that `stories` is moved from `exclude` to `include`

File: react-components/src/components/RevealContainer/RevealContext.ts [hunk 2]
```diff
@@ -1,12 +1,15 @@
...
+import { type Cognite3DViewer } from '@cognite/reveal';
+import { createContext, useContext } from 'react';
 
 export const RevealContext = createContext<Cognite3DViewer | null>(null);
 
-export const useReveal = () => {
+export const useReveal = (): Cognite3DViewer => {
   const reveal = useContext(RevealContext);
-  if (!reveal) {
+  if (reveal == null) {
```
Comment (haakonflatval-cognite) [inline react-components/src/components/RevealContainer/RevealContext.ts:11]:
```suggestion
  if (reveal === null) {
```

Should there be a lint rule for this?

File: react-components/src/components/RevealContainer/RevealContainer.tsx [hunk 3]
```diff
@@ -1,43 +1,55 @@
...
+import { useEffect, useRef, type ReactNode, useState, type ReactElement } from 'react';
 import { Cognite3DViewer } from '@cognite/reveal';
-import { RevealContext } from "./RevealContext";
-import { Color } from "three";
+import { RevealContext } from './RevealContext';
+import { type Color } from 'three';
 
-type RevealContainerProps = {
+interface RevealContainerProps {
   color?: Color;
```
Comment (haakonflatval-cognite) [inline react-components/src/components/RevealContainer/RevealContainer.tsx:11]: Will the lint rule prefer `interface` over `type`, or is this change just a matter of taste?

File: react-components/src/components/Image360HistoricalDetails/utils/FormatDate.ts [hunk 4]
```diff
@@ -2,8 +2,8 @@
  * Copyright 2023 Cognite AS
  */
 
-export const formatDate = (date: Date | undefined) => {
-  if (!date) {
+export const formatDate = (date: Date | undefined): string => {
+  if (date == null) {
```
Comment (haakonflatval-cognite) [inline react-components/src/components/Image360HistoricalDetails/utils/FormatDate.ts:6]:
```suggestion
  if (date === null) {
```

**Review by haakonflatval-cognite — APPROVED** [review]
LGTM, mostly :+1:

File: react-components/tsconfig.json (hunk 1 above)
Comment (christjt) [reply react-components/tsconfig.json:18]: ts gets mad at me if I don't add this :(

**Pull request merged**